user_mail_client/imp_res_users.py

```diff
@@ -0,0 +1,347 @@
+        if passwd:
+            values['dovecot_password'] = self.generate_dovecot_sha512(passwd)
+            remote_user_id = self.remote_user(self)
+            if remote_user_id:
+                _logger.info("VALUES IN WRITE :::::::::: %s" % values)
+                # remote_user_id.write(values)
+        return super(res_users, self).write(values)
+
+    def unlink(self):
+        user_id = self.env['res.users'].sudo().search([('login', '=', self.login)]).id
+        self.env['postfix.alias'].search([('user_id', '=', user_id)]).unlink()
+        # only needed if deleting a user with recently changed password
+        self.env['change.password.wizard'].search([('user_ids', '=', self.id)]).unlink()
+        self.env['change.password.user'].search([('user_id', '=', self.id)]).unlink()
+
+        remote_user = self.remote_user(self)
+        if remote_user:
+            remote_user.unlink()
+
+        return super(res_users, self).unlink()
+
+    # def testxmlrpc(self):
+    #     # ~ common = xmlrpc.client.ServerProxy('http://localhost:8069/xmlrpc/2/common')
+    #     # ~ info = common.version()
+    #     # ~ uid = common.authenticate('mail_server', 'admin', 'admin', {})
+    #     # ~ model = xmlrpc.client.ServerProxy('http://localhost:8069/xmlrpc/2/object',allow_none=True)
+    #
+    #     # ~ res = model.execute_kw('mail_server', uid, 'admin','res.company', 'search', [[['remote_id','=',None]]])
+    #     # ~ raise Warning(res,info,uid)
+    #     # ~ #_logger.error(res)
+    #
+    #     SYNCSERVER = Sync2server(self)
+    #     # ~ record = {f:self.read()[0][f] for f in self.USER_MAIL_FIELDS}
+    #     # ~ record['postfix_alias_ids'] = [(0,0,{'name': m.name, 'mail':m.mail,'active':m.active}) for m in self.postfix_alias_ids]
+    #
+    #     # ~ res = SYNCSERVER.sock.execute_kw(SYNCSERVER.passwd_dbnameSYNCSERVER.passwd_dbname, SYNCSERVER.uid, SYNCSERVER.passwd_passwd,'res.users', 'search', [[['id','=',1]]])
+    #     res = SYNCSERVER.sock.execute_kw('mail_server', 1, 'admin', 'res.users', 'search', [['id', '=', 1]])
+    #     _logger.error(res)
+    #
+    #     remote_company = SYNCSERVER.search('res.company', [['remote_id', '=', None]])
+    #
+    #     remote_company = SYNCSERVER.remote_company(self.company_id)
+    #
+    #     # raise Warning(uid,info)
+
+
+class res_company(models.Model):
+    _inherit = 'res.company'
+
+    def _get_param(self, param, value):
+        if not self.env['ir.config_parameter'].get_param(param):
+            self.env['ir.config_parameter'].set_param(param, value)
+        return self.env['ir.config_parameter'].get_param(param)
+
+    def generateUUID(self):
+        return str(uuid.uuid4())
+
+    def write(self, values):
+        if not self.remote_id:
+            values['remote_id'] = self.generateUUID()
+        if values.get('domain', False):
+            _logger.info('::::::::::values - %s', self)
+            # self.company_sync_settings(self)
```

Comment:
Commented code
Is this code purposely commented?
Should it be kept, is so motivate with a comment
Otherwise remove.

Reply:
yes, commented purposely, but I have removed it